src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -294,6 +294,10 @@ public static function isIgnoringTouch($class = null)
     {
         $class = $class ?: static::class;
 
+        if (! get_class_vars($class)['timestamps']) {
```

Comment:
I would prefer `(new $class)->usesTimestamps()` to be consistent with existing code.

Reply:
> We also have to cover cases where timestamps are enabled, but UPDATED_AT is null:
just added test

Reply:
@staudenmeir in some cases we're getting abstract classes here for ex.
https://github.com/laravel/framework/blob/5992f10116f612d51465385b0af11d778d975b87/tests/Database/DatabaseEloquentIntegrationTest.php#L1539

Reply:
Yes, but only in the tests. You can't get an abstract class in a real application. Please adjust the test accordingly.

Reply:
> Yes, but only in the tests. You can't get an abstract class in a real application.

OK, but If I use `(new $class)->usesTimestamps()` 6 more test are failing (not including mine) - what to do with that?
For example this one:
https://github.com/laravel/framework/blob/5992f10116f612d51465385b0af11d778d975b87/tests/Database/DatabaseEloquentIntegrationTest.php#L1539

Reply:
I think we should move the new code to `Relation::touch()`. `Model::isIgnoringTouch()` is only about `$ignoreOnTouch`, not about timestamps.